src/domain/session/room/timeline/deserialize.js

```diff
@@ -0,0 +1,398 @@
+import { MessageBody, HeaderBlock, ListBlock, CodeBlock, PillPart, FormatPart, NewLinePart, RulePart, TextPart, LinkPart, ImagePart } from "./MessageBody.js"
+import { linkify } from "./linkify/linkify.js";
+import { parsePillLink } from "./pills.js"
+
+/* At the time of writing (Jul 1 2021), Matrix Spec recommends
+ * allowing the following HTML tags:
+ *     font, del, h1, h2, h3, h4, h5, h6, blockquote, p, a, ul, ol, sup, sub, li, b, i, u,
+ *     strong, em, strike, code, hr, br, div, table, thead, tbody, tr, th, td, caption, pre, span, img
+ */
+
+/**
+ * Nodes that don't have any properties to them other than their tag.
+ * While <a> has `href`, and <img> has `src`, these have... themselves.
+ */
+const basicInline = ["EM", "STRONG", "CODE", "DEL", "SPAN" ];
+const basicBlock = ["DIV", "BLOCKQUOTE"];
+
+class Deserializer {
+    constructor(result, mediaRepository) {
+        this.result = result;
+        this.mediaRepository = mediaRepository;
+    }
+
+    parseLink(node, children) {
+        const href = this.result.getAttributeValue(node, "href");
+        if (!href || !href.match(/^[a-z]+:[\/]{2}/i)) {
```

Comment:
Relative URLs are thrown away here. Of note, though, is that Element allows relative URLs.